docs/src/content/hardhat-runner/docs/guides/typescript.md

```diff
@@ -10,188 +10,131 @@ Hardhat will automatically enable its TypeScript support if your config file end
 
-  it("should do something right", async function () {
-    // Do something with the accounts
-  });
-});
-```
+For example, the sample project's config turns from this:
 
-An example for scripts:
+<<< @/../packages/hardhat-core/sample-projects/javascript/hardhat.config.js{1}
 
-```typescript
-import { run, ethers } from "hardhat";
+into this:
 
-async function main() {
-  await run("compile");
+<<< @/../packages/hardhat-core/sample-projects/typescript/hardhat.config.ts{1,2,4-6}
 
-  const accounts = await ethers.getSigners();
+Finally, you need to create a [`tsconfig.json`](https://www.typescriptlang.org/docs/handbook/tsconfig-json.html) file. Here's our recommended one:
 
-  console.log(
-    "Accounts:",
-    accounts.map((a) => a.address)
-  );
-}
+<<< @/../packages/hardhat-core/sample-projects/typescript/tsconfig.json
 
-main()
-  .then(() => process.exit(0))
-  .catch((error) => {
-    console.error(error);
-    process.exit(1);
-  });
-```
+And that's really all it takes. Now you can write your config, tests, tasks and scripts in TypeScript.
 
-## Type-safe smart contract interactions
+## Writing tests and scripts in TypeScript
 
-If you want to type-check smart contract interactions (calling methods, reading events), use [`@typechain/hardhat`](https://github.com/ethereum-ts/TypeChain/tree/master/packages/hardhat). It generates typing files (`*.d.ts`) based on ABI's, and it requires little to no configuration when used with Hardhat.
+When using JavaScript, all the properties in the [Hardhat Runtime Environment](../advanced/hardhat-runtime-environment.md) are injected into the global scope. When using TypeScript nothing will be available in the global scope and you will need to import everything explicitly using, for example, `import { ethers } from "hardhat"`.
 
-## Type-safe configuration
+Follow the [Getting started guide](../getting-started/index.md) and create a TypeScript project for a complete example on how to write tests and scripts using TypeScript.
 
-One of the advantages of using TypeScript, is that you can have a type-safe configuration, and avoid typos and other common errors.
+## Type-safe smart contract interactions
 
-To do that, you have to write your config in this way:
+:::tip
 
-```ts
-import { HardhatUserConfig } from "hardhat/config";
+If you installed [`@nomicfoundation/hardhat-toolbox`](../../plugins/nomicfoundation-hardhat-toolbox) you can skip this section, as it includes [`@typechain/hardhat`](https://github.com/ethereum-ts/TypeChain/tree/master/packages/hardhat).
 
-const config: HardhatUserConfig = {
-  // Your type-safe config goes here
-};
+:::
 
-export default config;
-```
+If you want Hardhat to generate types for your smart contract you should install use [`@typechain/hardhat`](https://github.com/ethereum-ts/TypeChain/tree/master/packages/hardhat). It generates typing files (`*.d.ts`) based on ABI's, and it requires little to no configuration.
```

Comment:
```suggestion
If you want Hardhat to generate types for your smart contract you should install [`@typechain/hardhat`](https://github.com/ethereum-ts/TypeChain/tree/master/packages/hardhat). It generates typing files (`*.d.ts`) based on ABI's, and it requires little to no configuration.
```

Reply:
Or maybe you meant `install and use`? Otherwise this might seem like only installing typechain is enough.